docs/general/use-cases/pattern-matching/speeding-up-pcre2-with-sljit.md

```diff
@@ -0,0 +1,84 @@
+# Speeding Up PCRE2 With SLJIT
+
+[PCRE2](https://www.pcre.org/) is a C library for matching regular expressions. This library translates patterns into byte code, and this byte code can be executed by its interpreter. The library also has a just-in-time (JIT) compiler which can produce machine code from the byte code to further improve the matching performance. Before going into the details of the JIT compiler, the reason of choosing byte code as an internal representation is explained first.
+
+Regular expressions were originally designed for defining a set of words using special patterns, and engines only needed to decide whether a given word is in the set. Eventually patterns become a list of commands, and pattern matching engines become script languages. Today many constructs of structured programming are available in regular expressions (pcre2pattern / perlre) such as loops (called repetitions), if-else selection (called conditional blocks), function calls (called recursions), returns and breaks (called control verbs). The execution of each command is accurately defined which allows fine control over the entire matching process. Patterns can also store information in local variables (called capturing blocks and marks) and these variables are often available after the matching is completed. Byte code is used as an internal representation in many script languages, and modern regular expression engines are no different. There are several methods to improve the performance of byte code execution (including JIT compiling), but even the fastest script language can do very little to improve the speed of a slow bubble sort. Hence it is recommended to carefully design patterns to avoid slow matching speed.
```

Comment:
This looks like a very long line, kind of hard to change it in the future. Is this the preferred way to .md?